ingestion/src/metadata/utils/sql_lineage.py

```diff
@@ -0,0 +1,165 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Helper functions to handle SQL lineage operations
+"""
+import traceback
+from logging.config import DictConfigurator
+
+from metadata.generated.schema.api.lineage.addLineage import AddLineageRequest
+from metadata.generated.schema.entity.data.table import Table
+from metadata.generated.schema.type.entityLineage import EntitiesEdge
+from metadata.generated.schema.type.entityReference import EntityReference
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.utils import fqn
+from metadata.utils.helpers import _get_formmated_table_name
+from metadata.utils.logger import utils_logger
+
+
+# Prevent sqllineage from modifying the logger config
+def configure(self):
+    pass
+
+
+DictConfigurator.configure = configure
+
+
+logger = utils_logger()
+
+
+def _separate_fqn(database, fqn):
+    database_schema, table = fqn.split(".")[-2:]
+    if not database_schema:
+        database_schema = None
+    return {"database": database, "database_schema": database_schema, "name": table}
+
+
+def _create_lineage_by_table_name(
+    metadata: OpenMetadata,
+    from_table: str,
+    to_table: str,
+    service_name: str,
+    database: str,
+):
+    """
+    This method is to create a lineage between two tables
+    """
+    try:
+        from_table = str(from_table).replace("<default>", "")
+        to_table = str(to_table).replace("<default>", "")
+        # TODO: FIXME
+        from_fqdn = fqn.build(
```

Comment:
the value of `from_table` & `to_table` depends on the query we get while running the usage workflow.

when we get query as 
```
select * from info_schema.new_table
```
`from_table` value becomes `info_schema.new_table`

when query is 
```
select * from new_table
```
`from_table` value becomes `<default>.new_table`

and in above code i'm replacing the `<default>` with empty string, so we can split string with '.' and should get the schema & table name 

when the schema is empty string we use elastic search to fetch table entity.